### ISSUE TITLE
Bundle fails on accessing package config .json.json

### ISSUE THREAD

**Issue by mikz** [systemjs/builder#508]
I'm doing jspm bundle and it fails on loading `upsta/jspm_packages/npm/deku@2.0.0-rc16.json.json`.

This can be reproduced in the browser by:
```
System.import('jspm_packages/npm/deku@2.0.0-rc16.json')
```

It tried to load the package config, because https://github.com/systemjs/builder/blob/bb09bcb6a3911ae2abcc8492b2797ebd2006705c/lib/trace.js#L511-L530
is called with `tracePackageConfig` set to true, so `Trace.getLoadDependencies` returns the packageConfig as a dependency.

That happens because `traceCanonical` is called with: `{ tracePackageConfig: true }`.
So I guess that  `tracePackageConfig` is set to true is intentional.
That would explain why build works, but bundle doesn't. 

Disabling tracePackageConfig on https://github.com/systemjs/builder/blob/bb09bcb6a3911ae2abcc8492b2797ebd2006705c/lib/builder.js#L529 fixes the issue for me.

But digging deeper:

getPackageConfigPath for 
`file:///Users/mikz/Developer/UPSTA/app/assets/javascripts/jspm_packages/npm/deku@2.0.0-rc16.json` is `file:///Users/mikz/Developer/UPSTA/app/assets/javascripts/jspm_packages/npm/deku@2.0.0-rc16.json.json`

Given my configuration is:
```json
{
  "packageConfigPaths": [
    "npm:@*/*.json",
    "npm:*.json",
    "github:*/*.json"
  ]
}
```

So inside the `getPackageConfigPath`:

`file:///Users/mikz/Developer/UPSTA/app/assets/javascripts/jspm_packages/npm/deku@2.0.0-rc16.json` gets matched against `/^(file:\/\/\/Users\/mikz\/Developer\/UPSTA\/app\/assets\/javascripts\/jspm_packages\/npm\/[^\/]+)(\/|$)/`
and adds the .json suffix.

Changing https://github.com/systemjs/builder/blob/bb09bcb6a3911ae2abcc8492b2797ebd2006705c/lib/utils.js#L242

from:

```js
      configPath = pkgName + packageConfigPath.substr(p.length);
```

to:
```js
      var suffix = packageConfigPath.substr(p.length)
      configPath = pkgName.endsWith(suffix)? pkgName : pkgName + suffix
```

Fixes the problem for me, but no idea if it is good solution.
The commit is in https://github.com/mikz/builder/commit/823837bb1c2c7fc29486ba3b1e57e05f9b3ebe7f

**Comment by guybedford** [comment]
@mikz for this reason there is nowhere in SystemJS that does `System.import('package.json')` because it means the package name is "package.json" so actually loading package.json.json is correct unless we ban package names ending in .json, which would be silly. Instead we use `System.load('http://url/to/package.json')` in the package system which is a pre-normalize import.

How specifically are you getting this error in the builder? That would help to know.

**Comment by mikz** [comment]
@guybedford would you be ok with adding you as a collaborator of the project? I started it few days ago and I'm not ready yet to have it public. I have there a revisions that fails.

I had some misconfiguration there, because both the package and the project had the same name, which resulted in some weird behaviour. Like that after bundling, all packages started with `upsta/jspm_packages`.  It might be better to hard crash in that case, so if you are interested I can give you access and a revision that fails. Otherwise, we can close this.

**Comment by guybedford** [comment]
If the replication is very simple I'm more than happy to look into it, but I can't put a lot of time into diagnosing the issue from your code - the most you can help point me directly to it the better.

**Comment by guybedford** [comment]
* the more! But it really does help to know config edge cases, so it's nice to catch these.

**Comment by mikz** [comment]
@guybedford done. The branch is https://github.com/mikz/UPSTA/compare/jspm-systemjs-builder-issue
And the commit that breaks it is again is https://github.com/mikz/UPSTA/commit/1d968f5e39fa70108788b9c224f51250526d9c66
And the message has a step to reproduce.

**Comment by guybedford** [comment]
This was actually quite an intricate bug. Fixed in https://github.com/systemjs/builder/commit/2363880261a914045c9fa919bba7d00cf218f453. Thanks so much for the test case here.

**Comment by mikz** [comment]
@guybedford wow! Thanks!

**Comment by guybedford** [comment]
Released in 0.15.10.
